tests/integration/ossm_installer_int_test.go

```diff
@@ -89,6 +93,96 @@ var _ = When("Migrating Data Science Projects", func() {
 
 })
 
+var _ = When("Checking for CRD", func() {
+	var (
+		ossmInstaller *ossm.OssmInstaller
+	)
+
+	BeforeEach(func() {
+		ossmInstaller = ossm.NewOssmInstaller(&kfconfig.KfConfig{}, envTest.Config)
+	})
+
+	It("should successfully check existing CRD", func() {
+		// given example CRD installed into env from /ossm/test/crd/
+		crdGroup := "ossm.plugins.kubeflow.org"
+		crdVersion := "test-version"
+		crdResource := "test-resources"
+
+		// when
+		err := ossmInstaller.VerifyCRDInstalled(crdGroup, crdVersion, crdResource)
+
+		// then
+		Expect(err).ToNot(HaveOccurred())
+	})
+
+	It("should fail to check non-existing CRD", func() {
+		// given
+		crdGroup := "non-existing-group"
+		crdVersion := "non-existing-version"
+		crdResource := "non-existing-resource"
+
+		// when
+		err := ossmInstaller.VerifyCRDInstalled(crdGroup, crdVersion, crdResource)
+
+		// then
+		Expect(err).To(HaveOccurred())
+		Expect(err.Error()).To(ContainSubstring("server could not find the requested resource"))
+	})
+})
+
+var _ = When("Ensuring environment is set up correctly", func() {
+
+	var (
+		objectCleaner *testenv.Cleaner
+		ossmInstaller *ossm.OssmInstaller
+		name          = "test-name"
+		namespace     = "test-namespace"
+	)
+
+	BeforeEach(func() {
+		ossmInstaller = ossm.NewOssmInstaller(&kfconfig.KfConfig{}, envTest.Config)
+		objectCleaner = testenv.CreateCleaner(cli, envTest.Config, timeout, interval)
+	})
+
+	It("should return status if SMCP is found and status is available", func() {
+		ns := createNamespace(namespace)
+		Expect(cli.Create(context.Background(), ns)).To(Succeed())
+		smcpObj := &unstructured.Unstructured{
+			Object: map[string]interface{}{
+				"apiVersion": "maistra.io/v1",
+				"kind":       "ServiceMeshControlPlane",
+				"metadata": map[string]interface{}{
+					"name":      name,
+					"namespace": namespace,
+				},
+				"spec": map[string]interface{}{},
+			},
+		}
+		createErr := CreateSMCP(namespace, smcpObj, ossmInstaller.GetClientConfig())
```

Comment:
I think you can get the config from testenv instead. So there will be no need to exposing it through `GetClientConfig` just to make tests happy.

Reply:
perfect, thanks [762f693](https://github.com/maistra/opendatahub-operator/pull/7/commits/762f693754b366dae926703661a23a92e22362f4)